Sprint-1/fix/median.js

```diff
@@ -6,8 +6,26 @@
 // or 'list' has mixed values (the function is expected to sort only numbers).
 
 function calculateMedian(list) {
-  const middleIndex = Math.floor(list.length / 2);
-  const median = list.splice(middleIndex, 1)[0];
+  // Throw out invalid arrays
+  if (!Array.isArray(list)) {
+    return null;
+  }
+
+  // Filter and sort numbers only
+  const sortedList = list
+    .filter((item) => typeof item === "number" && !isNaN(item))
+    .sort((a, b) => a - b);
+
+  // Throw out empty arrays
+  if (sortedList.length < 2) {
```

Comment:
This is an interesting check - how did you decide `sortedList.length < 2` is a good check here? Why not 3, or 1?

Is there a test we  could add to the test suite to show we've done the right thing?

Reply:
I agree, it should really be `sortedList.length < 1`. I've checked and the function can handle arrays with only 1 entry.

I've updated the function and added a test to check sing entry arrays.